tests/integration/integration_test_runner.sh

```diff
@@ -0,0 +1,223 @@
+#!/bin/bash
+
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+BLUE='\033[0;36m'
+NC='\033[0m'
+
+DATA_DIR="data"
+SUCCESS_COUNT=0
+FAILURE_COUNT=0
+
+# free port for dummy API
+lsof -ti tcp:5000 | xargs kill &> /dev/null
+
+# start the dummy API
+printf "%s\n" "" "    [TEST] Starting dummy APIs" ""
+python mock_dockerhub_api.py &> /dev/null &
+echo "done"
+
+cd ../..
+
+# run updater CLI
+printf "%s\n" "" "    [TEST] Running updater - no PR to create" ""
+rm -f last-diff-result.log
+source tests/integration/env.test && ONLY_UPDATE_TAGS_HISTORY=true ./main.sh
+
+# update an image
+printf "%s\n" "" "    [INFO] Pushing weaver tag to DockerHub" ""
+printf "${BLUE}"
+curl -s -XPOST localhost:5000/pavics/weaver/1.13.2-worker
+printf "${NC}"
+
+# run updater CLI
+printf "%s\n" "" "    [TEST] Running updater - need to create a PR for [bump_weaver-worker_to_1.13.2-worker]" ""
+rm -f last-diff-result.log
+source tests/integration/env.test && ./main.sh
+
+
+### Asserts that diff contains the right thing
+printf "%s\n" "" "    [TEST] ASSERT" ""
+if grep -q 'export WEAVER_WORKER_IMAGE="pavics/weaver:1.13.2-worker"' "last-diff-result.log"; then
+    printf "${GREEN}[INFO] [bump_weaver-worker_to_1.13.2-worker] The commit content looks good"
+    echo
+    echo
+    cat last-diff-result.log
+    printf "${NC}"
+    ((SUCCESS_COUNT++))
+else
+    printf "${RED}[ERROR] [bump_weaver-worker_to_1.13.2-worker] wrong commit content."
+    echo
+    echo
+    cat last-diff-result.log
+    printf "${NC}"
+    ((FAILURE_COUNT++))
+fi
+
+
+# run updater CLI
+printf "%s\n" "" "    [TEST] Running updater - no PR to create" ""
+rm -f last-diff-result.log
+source tests/integration/env.test && ./main.sh
+
+# update an image
+printf "%s\n" "" "    [INFO] Pushing weaver and finch tags to DockerHub" ""
+printf "${BLUE}"
+curl -s -XPOST localhost:5000/pavics/weaver/1.13.3-worker
+curl -s -XPOST localhost:5000/birdhouse/finch/version-0.5.4
```

Comment:
So here we create a new fake finch docker tag `version-0.5.4`, how does the updater know this is newer?  1 year from now, when running this test, the current finch version in `default.env` could be `0.7.8` for example.

Reply:
mock_data/birdhouse_finch seems to mock version up to 0.5.3, so I guess that it's ok.

Reply:
> So here we create a new fake finch docker tag `version-0.5.4`, how does the updater know this is newer? 1 year from now, when running this test, the current finch version in `default.env` could be `0.7.8` for example.

In this case, for the first time you run it after one year, `version-0.7.8` would become the new tag on which we do the diff (in `fetch_tags`), since `version-0.5.4` was our last one. Then the commit would be empty (no diff after replacing `version-0.7.8` with `version-0.7.8`), so no PR.

Then, when we would run it one more time, the diff between `version-0.7.8` and `version-0.7.8` (our last registered tag) would be null, so we would know that finch is at its latest version already.